src/usermod.c

```diff
@@ -328,12 +336,52 @@ static int get_groups (char *list)
 	 * Any errors in finding group names are fatal
 	 */
 	if (0 != errors) {
+		close_group_files ();
 		return -1;
 	}
 
+	close_group_files ();
```

Comment:
Would it make sense to avoid the duplication here and move the close before the if()?

Reply:
As you say it makes more sense to close the files before the if(), in order to minimize future errors.